platform/security/sts/security-sts-attributequeryclaimshandler/src/main/java/org/codice/ddf/security/claims/attributequery/AttributeQueryClaimsHandler.java

```diff
@@ -0,0 +1,271 @@
+import org.apache.cxf.sts.claims.ClaimsParameters;
+import org.apache.cxf.sts.claims.ProcessedClaim;
+import org.apache.cxf.sts.claims.ProcessedClaimCollection;
+import org.opensaml.saml2.core.Assertion;
+import org.opensaml.saml2.core.Attribute;
+import org.opensaml.saml2.core.AttributeStatement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.security.PropertiesLoader;
+import ddf.security.common.audit.SecurityLogger;
+import ddf.security.samlp.SimpleSign;
+
+public class AttributeQueryClaimsHandler implements ClaimsHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AttributeQueryClaimsHandler.class);
+
+    private String attributeMapLocation;
+
+    private List<String> supportedClaims;
+
+    private Map<String, String> attributeMap;
+
+    protected SimpleSign simpleSign;
+
+    protected String externalAttributeStoreUrl;
+
+    protected String issuer;
+
+    protected String destination;
+
+    protected static final String ERROR_RETRIEVING_ATTRIBUTES = "Error retrieving attributes from external attribute store [{}] for DN [{}]. ";
+
+    protected static final String ERROR_RETRIEVING_ATTRIBUTES_SEC_LOG = "Error retrieving attributes from external attribute store [%s] for DN [%s]. ";
+
+    public AttributeQueryClaimsHandler() {
+        LOGGER.debug("Starting AttributeQueryClaimsHandler");
+    }
+
+    @Override
+    public List<URI> getSupportedClaimTypes() {
+        LOGGER.debug("Getting supported claim types.");
+        List<URI> supportedClaimTypes = new ArrayList<>();
+        try {
+            for (String claim : supportedClaims) {
+                supportedClaimTypes.add(new URI(claim));
+            }
+        } catch (URISyntaxException e) {
+            LOGGER.error("Error getting supported claim types for external attribute store.", e);
+        }
+        return supportedClaimTypes;
+    }
+
+    @Override
+    public ProcessedClaimCollection retrieveClaimValues(ClaimCollection claims,
+            ClaimsParameters parameters) {
+        Principal principal = parameters.getPrincipal();
+
+        String username = getUsername(principal);
+
+        ProcessedClaimCollection claimCollection = null;
+
+        try {
+            if (!StringUtils.isEmpty(username)) {
+                claimCollection = retrieveExternalClaimValues(username);
+            }
+        } catch (URISyntaxException e) {
+            LOGGER.error(ERROR_RETRIEVING_ATTRIBUTES, externalAttributeStoreUrl, username, e);
+            SecurityLogger.logError(
+                    String.format(ERROR_RETRIEVING_ATTRIBUTES_SEC_LOG, externalAttributeStoreUrl,
+                            username), e);
+        }
+
+        return (claimCollection != null) ? claimCollection : new ProcessedClaimCollection();
+    }
+
+    /**
+     * Retrieves claims from the external attribute store.
+     *
+     * @param username of the user.
+     * @return The collection of claims.
```

Comment:
Personal preference: if a method can return a null value, its javadoc should indicate the conditions under which that happens.

e.g. _@return The collection of claims or null if there are no security claims._